### ISSUE TITLE
Error reading record# 5564 from unit# 31

### ISSUE THREAD

**Issue by wapmanzoo** [pixie16/paass#107]
Pixie_scan quits going through an ldf file with the following error:

```
Error reading record # 5564 from unit# 31
Number of buffers processed = 780
```

This is error from file september_001.ldf that was linked to google drive to @cthornsb, @spaulaus and @ksmith0 thru email.
This error has occurred for other ldf files scanned by pixie scan. The record# and buffers processed changes while "from unit# 31" is common to every occurrence.

This prevents scanning through the entire ldf file and is wasting a bunch of data.

**Comment by wapmanzoo** [comment]
I recorded 60 ldfs files over the weekend, each about 1 hours worth of data, about 450 buffers.
7 files crashed the scan code with the error in previous post.
Record was usually 457 or 458, and once record was 268.
All were from unit # 31.
Buffers processed was between 16 and 413.
This is wasting about 5% of my data right now, as I can't scan the effected files past these crash points.
Right now I'm going to skip these ldfs files altogether until the bug is fixed and the scan code learns to just skip to the next buffer.

**Comment by spaulaus** [comment]
Is there any update on this issue?

**Comment by wapmanzoo** [comment]
I still get the same error. I call it Unit31 error. It occurs between 3-20% of the ldfs files. To reduce the amount of lost data, I set the max file size small. So the 3-20% of errors in the ldfs means I lose only 1.5-10% of my data. If the files sizes were kept longer I'm confident that I'd be losing half or more of my data.

**Comment by spaulaus** [comment]
Did you ever provide one of these data files to us? I do not see this issue with data files that I have.

**Comment by wapmanzoo** [comment]
I did send out a smaller ldf that had this error. This was sent to Stan, Cory and Karl gmail accounts on Sept 22 2016. The ldf is titled triples_004.ldf

**Comment by ksmith0** [comment]
I've uploaded [triples_004.ldf](https://vandle.slack.com/files/ksmith0/F3XMQBFD5/triples_004.ldf) to the VANDLE Slack [#fissionstudyexp](https://vandle.slack.com/archives/fissionstudyexp) channel.

**Comment by ksmith0** [comment]
This may have been resolved in commit b5d2036ca0ac4370fbdb67ab5f8fe63e20a0d204. More information in #201.

**Comment by cthornsb** [comment]
This issue has been resolved in commit b5d2036. Test files taken before this commit (with a polling threshold of 0.1%) exhibit the same problem when scanned with pixie_scan. @wapmanzoo tested the new commit with a polling threshold of 1% on NITemp and was able to scan all files taken.
